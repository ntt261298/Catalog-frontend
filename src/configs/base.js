export default {
  apiHost: 'http://localhost:5000',
};
