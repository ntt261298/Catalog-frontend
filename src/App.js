import React, { Component } from 'react';
import Header from './components/layout/Header';
import Home from './components/home';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <Home />
      </div>
    );
  }
}

export default App;
