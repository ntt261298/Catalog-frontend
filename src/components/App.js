import React from 'react';
import {
  BrowserRouter as Router, Route, Redirect, Switch,
} from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import HomePage from 'components/Home/HomePage';
import CategoryPage from 'components/Category/CategoryPage';
import ItemPage from 'components/Item/ItemPage';
import UserPage from 'components/User/UserPage';
import Page404 from 'components/Layout/404';

export const App = props => (
  <Router>
    <div className="App">
      <Switch>
        <Route path="/" exact component={HomePage} />
        <Route path="/category/:id" exact component={CategoryPage} />
        <Route path="/category/:categoryID/item/:itemID" exact component={ItemPage} />
        <Route path="/me/items" exact render={() => (!props.loggedIn ? <Redirect to="/" /> : <UserPage />)} />
        <Route component={Page404} />
      </Switch>
    </div>
  </Router>
);

App.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  loggedIn: state.user.loggedIn,
});

export default connect(mapStateToProps, null)(App);
