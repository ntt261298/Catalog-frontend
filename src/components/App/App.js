import React from 'react';
import {
  BrowserRouter as Router, Route, Redirect, Switch,
} from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import HomePage from '../Home/HomePage';
import CategoryPage from '../Category/CategoryPage';
import ItemPage from '../Item/ItemPage';
import UserPage from '../User/UserPage';
import Page404 from './404';

export const App = props => (
  <Router>
    <div className="App">
      <Switch>
        <Route path="/" exact component={HomePage} />
        <Route path="/category/:id" exact component={CategoryPage} />
        <Route path="/category/:categoryID/item/:itemID" exact component={ItemPage} />
        <Route path="/me/items" exact render={() => (!props.loggedIn ? <Redirect to="/" /> : <UserPage />)} />
        <Route component={Page404} />
      </Switch>
    </div>
  </Router>
);

App.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  loggedIn: state.user.loggedIn,
});

export default connect(mapStateToProps, null)(App);
