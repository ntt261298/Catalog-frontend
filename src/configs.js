import {
  configure, shallow, mount, render,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Enzyme config
configure({ adapter: new Adapter() });

// API host config
const defaultHost = 'http://localhost:5000';

export {
  shallow, mount, render, defaultHost,
};
