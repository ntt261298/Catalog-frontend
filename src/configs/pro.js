import base from 'configs/base';

export default {
  ...base,
};
