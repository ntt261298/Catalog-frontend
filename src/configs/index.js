import {
  configure, shallow, mount, render,
} from 'enzyme/build';
import Adapter from 'enzyme-adapter-react-16/build';
import dev from 'configs/dev';
import pro from 'configs/pro';

// Enzyme config
configure({ adapter: new Adapter() });

let configs;

if (process.env.REACT_APP_ENV === 'dev') {
  configs = dev;
} else if (process.env.REACT_APP_ENV === 'pro') {
  console.log('pro');
  configs = pro;
} else {
  // Default configs is dev
  configs = dev;
}

export { shallow, mount, render };

export default configs;
