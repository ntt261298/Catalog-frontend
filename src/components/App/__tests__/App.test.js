import React from 'react';
import { shallow } from 'configs';
import { App } from 'components/App/App';

describe('components/App/App', () => {
  let wrapper;
  let props;

  const setup = () => {
    wrapper = shallow(
      <App {...props} />,
    );
  };
  beforeEach(() => {
    props = {
      loggedIn: true,
    };
  });

  it('should render App', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });

  it('should render App with Redirect', () => {
    props.loggedIn = false;
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });
});
