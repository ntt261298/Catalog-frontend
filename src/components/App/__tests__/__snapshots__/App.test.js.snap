// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/App/App should render App 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/category/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/category/:categoryID/item/:itemID"
      />
      <Route
        exact={true}
        path="/me/items"
        render={[Function]}
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;

exports[`components/App/App should render App with Redirect 1`] = `
<BrowserRouter>
  <div
    className="App"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/category/:id"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/category/:categoryID/item/:itemID"
      />
      <Route
        exact={true}
        path="/me/items"
        render={[Function]}
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;
